language/move-lang/src/to_bytecode/translate.rs

```diff
@@ -0,0 +1,1022 @@
+    let mut context = Context::new(&ident, struct_declarations, function_declarations)?;
+
+    let mut struct_defs = vec![];
+    let mut field_defs = vec![];
+    for (s, sdef) in mdef.structs.into_iter() {
+        struct_def(
+            &mut context,
+            &mut struct_defs,
+            &mut field_defs,
+            &ident,
+            s,
+            sdef,
+        )?;
+    }
+
+    let function_defs = mdef
+        .functions
+        .into_iter()
+        .map(|(f, fdef)| function(&mut context, &ident, f, fdef))
+        .collect::<Result<Vec<_>>>()?;
+
+    let MaterializedPools {
+        module_handles,
+        struct_handles,
+        function_handles,
+        type_signatures,
+        function_signatures,
+        locals_signatures,
+        identifiers,
+        user_strings,
+        byte_array_pool,
+        address_pool,
+    } = context.materialize_pools();
+    let compiled_module_mut = F::CompiledModuleMut {
+        module_handles,
+        struct_handles,
+        function_handles,
+        type_signatures,
+        function_signatures,
+        locals_signatures,
+        identifiers,
+        user_strings,
+        byte_array_pool,
+        address_pool,
+        struct_defs,
+        field_defs,
+        function_defs,
+    };
+    let compiled_module = compiled_module_mut
+        .freeze()
+        .map_err(|vm_status| vec![(ident.loc(), format!("VM ERROR: {:#?}", vm_status))])?;
+    Ok((ident.0.value.name.clone(), compiled_module))
+}
+
+fn main(
+    address: Address,
+    main_name: FunctionName,
+    fdef: G::Function,
+    struct_declarations: &HashMap<(ModuleIdent, StructName), (bool, Vec<F::Kind>)>,
+    function_declarations: &HashMap<(ModuleIdent, FunctionName), G::FunctionSignature>,
+) -> Result<F::CompiledScript> {
+    let loc = main_name.loc();
+    let current_module_ = ModuleIdent_ {
+        address,
+        name: ModuleName(sp(loc, F::self_module_name().to_owned().to_string())),
+    };
+    let current_module = ModuleIdent(sp(loc, current_module_));
+    let mut context = Context::new(&current_module, struct_declarations, function_declarations)?;
+
+    let main = function(&mut context, &current_module, main_name, fdef)?;
+
+    let MaterializedPools {
+        module_handles,
+        struct_handles,
+        function_handles,
+        type_signatures,
+        function_signatures,
+        locals_signatures,
+        identifiers,
+        user_strings,
+        byte_array_pool,
+        address_pool,
+    } = context.materialize_pools();
+    let compiled_script = F::CompiledScriptMut {
+        module_handles,
+        struct_handles,
+        function_handles,
+        type_signatures,
+        function_signatures,
+        locals_signatures,
+        identifiers,
+        user_strings,
+        byte_array_pool,
+        address_pool,
+        main,
+    };
+    compiled_script
+        .freeze()
+        .map_err(|vm_status| vec![(loc, format!("VM ERROR: {:#?}", vm_status))])
```

Comment:
```
error[E0277]: `std::vec::Vec<libra_types::vm_error::VMStatus>` doesn't implement `std::fmt::Display`
   --> language/move-lang/src/to_bytecode/translate.rs:276:73
    |
276 |         .map_err(|vm_status| vec![(ident.loc(), format!("VM ERROR: {}", vm_status))])?;
    |                                                                         ^^^^^^^^^ `std::vec::Vec<libra_types::vm_error::VMStatus>` cannot be formatted with the default formatter
    |
    = help: the trait `std::fmt::Display` is not implemented for `std::vec::Vec<libra_types::vm_error::VMStatus>`
    = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
    = note: required by `std::fmt::Display::fmt`
```
Is this a new addition?

Reply:
Nope! It's been there since I updated the errors a couple months ago. I'm guess it's because it's a vector of `VMStatus` as opposed to a `VMStatus` that it's complaining. But I wouldn't waste any time over it unless you really want to.